src/main/java/com/gluonhq/substrate/target/IosTargetConfiguration.java

```diff
@@ -117,15 +122,18 @@ public IosTargetConfiguration(ProcessPaths paths, InternalProjectConfiguration c
                 "-isysroot", getSysroot());
     }
 
-    @Override
     List<String> getTargetSpecificAOTCompileFlags() throws IOException {
-        return Arrays.asList("-H:CompilerBackend=" + Constants.BACKEND_LLVM,
-                "-H:-SpawnIsolates",
+        List<String> flags = new ArrayList<>(Arrays.asList(
                 "-H:PageSize=16384",
                 "-Dsvm.targetName=iOS",
                 "-Dsvm.targetArch=" + getTargetArch(),
                 "-H:+UseCAPCache",
-                "-H:CAPCacheDir=" + getCapCacheDir().toAbsolutePath().toString());
+                "-H:CAPCacheDir=" + getCapCacheDir().toAbsolutePath().toString(),
+                "-H:CompilerBackend=" + projectConfiguration.getBackend()));
```

Comment:
Given that the current recommended Gluon's GraalVM release only allows LLVM backend for iOS, we are setting the LLVM backend if `projectConfiguration::backend` is not set:
```
if (projectConfiguration.getBackend() == null) {
         projectConfiguration.setBackend(Constants.BACKEND_LLVM);
 }
```
So this change `"-H:CompilerBackend=" + Constants.BACKEND_LLVM` -> `"-H:CompilerBackend=" + projectConfiguration.getBackend())` produces the same result as before, but also allows any change if the env var is set with any value (LIR or LLVM or other).

Since there is no public API for `projectConfiguration::backend`, changing the backend only can be done, after this PR, via `GRAALVM_BACKEND`.

I take that this is the same as:
> better not to change the default unless we want it
isn't it?

Reply:
I actually mean the GraalVM default. I would skip the -H:CompilerBackend flag as much as possible, but I'm ok with this too.

Reply:
It seems that only when "LLVM" is set, there are some implications:

https://github.com/oracle/graal/blob/master/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/SubstrateOptions.java#L386
and
https://github.com/oracle/graal/blob/master/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/SubstrateOptions.java#L407

Setting the flag with other values ("lir" is the default one, but null or any random value can be set) does not have any implication.